src/transform/src/canonicalization/flat_map_elimination.rs

```diff
@@ -7,16 +7,19 @@
 // the Business Source License, use of this software will be governed
 // by the Apache License, Version 2.0.
 
-//! Turns `FlatMap` into `Map` if only one row is produced by flatmap.
-//!
+//! For a `FlatMap` whose args are all constants, turns it into `Map` if only 1 row is produced by
+//! the table function, or turns it into an empty constant if 0 rows are produced by the table
+//! function. Additionally, a `Wrap` whose width is larger than its number of arguments can be
```

Comment:
I have no idea what `Wrap` does, but it does now feel like the odd function out. The explanation is a bit underdeveloped. If I were writing it, I think I'd make this a new paragraph, as it seems unrelated to the previous thought (it is not an instance of constant arguments, but rather an unrelated thing we can do with flat maps).

Also maybe just a nit, but the previous sentences are active voiced (this transform will do a thing) vs "can be removed" which .. does this transform do that or not? (( yes, but let's say that it does. ))

Reply:
Right, thanks!